### DISCUSSION TITLE
Setting height and width of rows and columns in dp.DataTable

### DISCUSSION THREAD

**Discussion by andrewhong5297** [datapane/datapane#94] [Help]
Does anyone know if there is a way to change default row/column widths in DataTable obj? One of my columns is a text column, so I want to make it easier to read the text there. 

Thanks!

**Comment by johnmicahreid** [comment]
Hi Andrew, you should be able to set the column width by applying a style to the Pandas DF and passing that into the Datatable. Try these out: 
- https://stackoverflow.com/questions/39680147/can-i-set-variable-column-widths-in-pandas
- https://docs.datapane.com/reports/blocks/tables-and-data

**Reply by Davidyang1989** [reply]
after applying a style to the Pandas DF and passing that into the Datatable like:

dp.DataTable(tableT.style.set_properties(subset=['column'], **{'width': '300px'}))



it returned error:
ValueError: Must return a primitive, pd.DataFrame, pd.Series or numpy array.

**Reply by mands** [reply]
Hi @Davidyang1989 - sorry styling is only supported when using `dp.Table`, which is a static table, and not `dp.DataTable`, which is a more interactive table component with sorting and filtering.

We're looking at adding more customisation to `dp.DataTable` to include setting column widths so longer columns can be viewed more easily
